src/app/api/collections/quality_matrix.py

```diff
@@ -314,34 +324,47 @@ def past_quality_matrix(
     return QualityMatrix.parse_obj(json.loads(result[0].quality_matrix))
 
 
-def quality_backup(session: Session):
-    logger.info("Storing quality matrices in database")
+def quality_backup(session: Session, timestamp: datetime.datetime):
+    """
+    Note: If multiple instances of the app are running (e.g. via
+    multiple gunicorn workers), we should not store the quality
+    matrix multiple times.
+
+    Hence, we pass in the timestamp of the scheduled save and use
+    it as part of the primary key. The database will then make sure
+    we cannot write duplicate instances.
+    """
+    logger.info(f"Storing quality matrices in database for {timestamp}")
+
+    modes: Tuple[QualityMatrixMode, ...] = ("replication-source", "collection")
```

Comment:
This is a duplicate definition of the literal. Please use an enum.

Reply:
Yes, this was an experiment how to simplify the Rest API (Literal instead of enum). I cant go back to an enum without again adapting the frontend code (which I'd rather avoid) in this PR.

The essence of this experiment: Using literal is less backend and less frontend code on the cost of a slighly less conveniant refactoring ("string based find replace" instead of "refactor rename"). It also avoids any headache about how the enum is serialized/deserialized via fastapi.

This could be done together with #103 where we currently also have one as a literal and one as enum

Yes it kinda is a duplication of the literal, unfortunately something like `QualityMatrixMode.values()` is not possible - with enum that would work :-(
However, your IDE actually tells you if you have invalid literals in that tuple (which is the whole idea of the Literal type annotation).

Reply:
Interesting. I am not ultra happy with the double string values, e.g., due to refactoring, but since we use good IDE's I expect them to highlight this. interesting new feature!